include/rocksdb/options.h

```diff
@@ -1946,6 +1946,20 @@ struct ReadOptions {
   // Default: false
   bool allow_unprepared_value = false;
 
+  // EXPERIMENTAL
+  //
+  // When enabled, the iterator will opportunistically check whether its current
+  // superversion number matches most recent CF counterpart during Seek(),
+  // Next() and Prev() calls. If not, it will refresh itself to the latest CF
+  // superversion preserving the very same snapshot it's been assigned to for
+  // consistency. This allows the iterator to adapt to updates such as new
+  // memtables, compactions and flushes.
+  //
+  // Note: This option requires iterator to have allow_refresh_ set to `true`.
```

Comment:
I believe the user-facing version of this is "Not supported for TransactionDB with WRITE_PREPARED or WRITE_UNPREPARED policies." It's also not applicable to SstFileReader but it wouldn't be useful for that anyway.

See https://github.com/facebook/rocksdb/commit/20dea55bb781ede1ba7e82484605be4d2beaaaf2 where the testsuite passes with only specific places allowed to use allow_refresh=false